core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -111,6 +113,12 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
         long lastValidKey = 0;
         int lastValidPosition = 0;
         int counter = 0;
+        double[] modelCoef = model.getCoef();
+
+        // Do we need to fit the model parameters to the data?
+        if (minimize && model.canBeMinimized()) {
```

Comment:
Can we add a check somewhere to throw an exception is `minimize` is `true` but `model.canBeMinimized()` returns `false`?